scripts/bootstrap.sh.in

```diff
@@ -150,8 +151,39 @@ orchestrate_bootstrap() {
         "$SALT_CALL" mine.update
 }
 
+
+bootstrap_file_is_present() {
+    if [ ! -f "$BOOTSTRAP_FILE" ]; then
+        die "$BOOTSTRAP_FILE is missing. You need to create a Bootstrap config" >&2
+        return 1
+    fi
+}
+
+validate_bootstrap_file() {
+    declare -a BootstrapConf=('apiVersion' 'kind' 'networks' 'ca' 'archives')
+    declare -a BootstrapConf_SubKeys=('controlPlane' 'workloadPlane' 'minion')
+    declare bootstrap_file_content
+    bootstrap_file_content=$(cat "${BOOTSTRAP_FILE}")
+
+    for key in "${BootstrapConf[@]}"; do
+        if [[ ! $bootstrap_file_content =~ $key ]]; then
+            die "$key is mandatory but cannot be found in $BOOTSTRAP_FILE"
+            return 1
+        fi
+    done
+
+    for subkey in "${BootstrapConf_SubKeys[@]}"; do
+        if [[ ! $bootstrap_file_content =~ $subkey ]]; then
+            die "$subkey is mandatory but cannot be found in $BOOTSTRAP_FILE"
+            return 1
+        fi
+    done
+}
```

Comment:
No, that kind of validation should be done with Salt, as we don't want to maintain two sources of such validation logic.
IMO. we want:

- to fix the code in `salt/_pillar/metalk8s.py` so that it doesn't fail with an `AssertionError` in the middle of its execution (instead, populate `_errors`  in the relevant pillar section)
- to fix the various formulas (SLS files) that rely on `pillar.metalk8s` to ensure it doesn't contain an `_errors` key when we must not have any (not sure where exactly, go go `grep`-o-gadget!)